apps/api/src/server/applications/application/documents/document.service.js

```diff
@@ -126,45 +146,20 @@ export const obtainURLsForDocuments = async (documentsToUpload, caseId) => {
 };
 
 /**
+ * Upserts the metadata for a document with the provided GUID using the provided metadata body.
  *
- * @param {number} id
+ * @param {string} documentGuid - The GUID of the document to upsert metadata for.
+ * @param {DocumentMetadata} documentMetadataBody - The metadata body to use for upserting.
+ * @returns {Promise<DocumentDetails>} A promise that resolves with the document details after the upsert.
  */
-export const fetchDocumentMetadata = async (id) => {
-	const documentMetadata = [
-		{
-			id: 77,
-			metadataId: 'abc123',
-			representative: 'DDE Agriculture',
-			documentGuid: 'xyz789',
-			caseRef: 'EN010120',
-			horizonDataID: 'EN010120-001',
-			version: '1.0',
-			filename: 'document.pdf',
-			originalFilename: null,
-			size: 1024,
-			mime: 'application/pdf',
-			documentURI: 'https://example.com/documents/abc123',
-			path: null,
-			virusCheckStatus: 'Scanned',
-			fileMD5: 'abcd1234',
-			dateCreated: '2023-02-21T10:30:00Z',
-			lastModified: '2023-02-21T11:30:00Z',
-			caseType: null,
-			documentStatus: 'Internal',
-			redacted: false,
-			published: false,
-			datePublished: null,
-			documentType: 'Report',
-			securityClassification: 'Official',
-			sourceSystem: 'NI Filestore',
-			origin: 'PINS',
-			owner: null,
-			author: null,
-			description:
-				'Attachments to the letter to Department for Business, Energy & Industrial Strategy',
-			stage: null
-		}
-	];
-
-	return documentMetadata.find((data) => data.id === id);
+export const upsertDocumentMetadataAndReturnDetails = async (
+	documentGuid,
+	documentMetadataBody
+) => {
+	const documentMetadata = await documentMetadataRepository.upsert({
+		...documentMetadataBody,
```

Comment:
I think we might be exposing a [mass assignment](https://cheatsheetseries.owasp.org/cheatsheets/Mass_Assignment_Cheat_Sheet.html) vulnerability here - looking at the joi schema, we allow the request to set things like the virus check status which I don't think the end user should ever be able to do.

This joi schema should describe the request model that the API allows (I guess this should just be details which the end user can modify).

Reply:
@chrisdalykinandcarta I did think that but I think we need to discuss a team what fields should be mutable to the end user and what can’t

Reply:
@chrisdalykinandcarta I also think at later point we need to add role management